cardano-cli/src/Cardano/CLI/EraBased/Transaction/Command.hs

```diff
@@ -260,12 +265,50 @@ data TransactionCalculateMinValueCmdArgs era = TransactionCalculateMinValueCmdAr
   }
   deriving Show
 
-data TransactionCalculatePlutusScriptCostCmdArgs = TransactionCalculatePlutusScriptCostCmdArgs
-  { nodeConnInfo :: !LocalNodeConnectInfo
+data TransactionCalculatePlutusScriptCostCmdArgs era = TransactionCalculatePlutusScriptCostCmdArgs
+  { nodeContextInfoSource :: !(NodeContextInfoSource era)
   , txFileIn :: FilePath
   , outputFile :: !(Maybe (File () Out))
   }
 
+-- | Either information about the context in which the transaction command
+-- is run, or information required to obtain it (information to connect to the node).
+data NodeContextInfoSource era
+  = NodeConnectionInfo !LocalNodeConnectInfo
+  | ProvidedTransactionContextInfo !(TransactionContext era)
+
+-- | Transaction context, required to evaluate the execution
+-- costs of the plutus scripts in the transaction.
+data TransactionContext era = TransactionContext
+  { shelleyBasedEra :: ShelleyBasedEra era
```

Comment:
This field appears unused. Can we remove it? I think `TransactionContext` does not need to be parameterised on era.

`era` in `File (UTxO era) In` is just informative. If it gives us compilation issues, maybe `File (Some UTxO) In` would be a better fit here?

This could also make `TransactionCalculatePlutusScriptCostCmdArgs` era independent, so maybe it should belong to the `EraIndependent` commands hierarchy?

I see that this command depends only on the node's era and the transaction's era, so I don't see any reason why it shouldn't.

Reply:
Where should we put it in the `EraIndependent` hierarchy? In `debug`?
The rest is addressed in: https://github.com/IntersectMBO/cardano-cli/pull/1079/commits/e25b52fcf1b340153f0618c49247d21e4287fbf0